### ISSUE TITLE
Missing features required by git

### ISSUE THREAD

**Issue by AtkinsSJ** [HeyPuter/puter#482]
The git client is limited by a few missing features in Puter, Terminal, and Phoenix. This issue exists to keep track of them.

- [ ] Request input from stdin, for username/password (#447)
- [ ] Launch an app to open a specific file. This is used for writing a commit message, and interactive rebase, among other things (#481)
- [ ] Add a CORS proxy, so that we don't have to rely on https://cors.isomorphic-git.org/
- [ ] Possibly add an SSH proxy? As far as I can tell, SSH can't be done directly from a web browser, but I could be wrong. This would be used for SSH authentication.

(I'll update this if and when I think of others. I'm sure there were more.)

**Comment by sysnajar** [comment]
Does Puter even have git client ? I've tried searching for it.

**Comment by AtkinsSJ** [comment]
> Does Puter even have git client ? I've tried searching for it.

It might not exist on puter.com yet but it's available on self-hosted.

**Comment by jelveh** [comment]
That's correct. The most cutting edge version of Puter is the open-source, self-hosted one which also has git right now.

**Comment by sysnajar** [comment]
great news , would try to docker version today. 
Had some issue launching terminal app on the dev version (WSL2).

**Comment by jelveh** [comment]
Could you elaborate on the issue you had trying to run dev? also, are you using Windows?

**Comment by sysnajar** [comment]
> Could you elaborate on the issue you had trying to run dev? also, are you using Windows?

Yes, I was gonna put it in issue tracking , but here's the quick one. (and yes I'm running it on ubuntu which is runs on windows11 's WSL2).

Steps to reproduce:
1. git clone & npm install , start
2. open the terminal app

i got the text Error /gui/ right on top of a white background instead of the expected terminal.
<img width="728" alt="puter" src="https://github.com/HeyPuter/puter/assets/3057714/9aec5d87-314b-41c9-93ae-0f82728fc0b2">

Thank you,
Owat

**Comment by jelveh** [comment]
Do other apps work fine?

**Comment by sysnajar** [comment]
> Do other apps work fine?

yes , every other apps I opened seems to work just fine.

**Comment by jelveh** [comment]
I wasn't able to reproduce this. Could you post a copy of what your terminal is displaying?
